3737/1/hash_of_gems.rb

```diff
@@ -0,0 +1,45 @@
+require_relative 'gem_info.rb'
+
+# class that collect all gems with their info
+class GemsCollection
+  def initialize
+    @gems = {}
+  end
+
+  def gems_list(option)
+    file = YAML.load_file(option[:file])
+    hash_of_gems(file['gems'])
+    select_options(option)
+  end
+
+  private
+
+  def hash_of_gems(file)
+    file.each do |lib|
+      stat = GemInfo.new(lib).gem_stat
+      next unless stat
+      @gems[lib] = stat
+    end
+    @gems
+  end
+
+  def select_options(option)
```

Comment:
are you sure that you're selecting options in this method? Please choose better name